src/runtime/virtual/server-handlers.d.ts

```diff
@@ -1,11 +1,17 @@
-import type { CompatibilityEventHandler, RouterMethod } from 'h3'
+import type { CompatibilityEventHandler, LazyEventHandler, RouterMethod } from 'h3'
 
-interface HandlerDefinition {
+type HandlerDefinition = {
   route: string
-  lazy?: boolean
   middleware?: boolean
-  handler: CompatibilityEventHandler | (() => Promise<CompatibilityEventHandler>)
   method?: RouterMethod
 }
+& {
+  lazy?: true
+  handler: LazyEventHandler | (() => Promise<LazyEventHandler>)
```

Comment:
We expect the handler still being path to a normal handler. `lazy: true`, does wrap dynamic import into lazy handler.